tests/scalers/prometheus/azure_managed_prometheus/azure_managed_prometheus_test.go

```diff
@@ -0,0 +1,331 @@
+  annotations:
+    prometheus.io/scrape: "true"
+spec:
+  ports:
+  - name: http
+    port: 80
+    protocol: TCP
+    targetPort: 8080
+  selector:
+    type: {{.MonitoredAppName}}
+`
+
+	azureManagedPrometheusWorkloadIdentityAuthTemplate = `apiVersion: keda.sh/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: azure-managed-prometheus-trigger-auth-wi
+  namespace: {{.TestNamespace}}
+spec:
+  podIdentity:
+      provider: azure-workload
+`
+
+	scaledObjectTemplate = `apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: {{.ScaledObjectName}}
+  namespace: {{.TestNamespace}}
+spec:
+  scaleTargetRef:
+    name: {{.DeploymentName}}
+  minReplicaCount: {{.MinReplicaCount}}
+  maxReplicaCount: {{.MaxReplicaCount}}
+  pollingInterval: 3
+  cooldownPeriod:  1
+  triggers:
+  - type: prometheus
+    metadata:
+      serverAddress: {{.PrometheusQueryEndpoint}}
+      metricName: http_requests_total
+      threshold: '20'
+      activationThreshold: '20'
+      query: sum(rate(http_requests_total{app="{{.MonitoredAppName}}"}[2m]))
+    authenticationRef:
+      name: azure-managed-prometheus-trigger-auth-wi
+`
+
+	generateLowLevelLoadJobTemplate = `apiVersion: batch/v1
+kind: Job
+metadata:
+  name: generate-low-level-requests-job
+  namespace: {{.TestNamespace}}
+spec:
+  template:
+    spec:
+      containers:
+      - image: quay.io/zroubalik/hey
+        name: test
+        command: ["/bin/sh"]
+        args: ["-c", "for i in $(seq 1 60);do echo $i;/hey -c 5 -n 30 http://{{.MonitoredAppName}}.{{.TestNamespace}}.svc;sleep 1;done"]
+        securityContext:
+          allowPrivilegeEscalation: false
+          runAsNonRoot: true
+          capabilities:
+            drop:
+              - ALL
+          seccompProfile:
+            type: RuntimeDefault
+      restartPolicy: Never
+  activeDeadlineSeconds: 100
+  backoffLimit: 2
+  `
+
+	generateLoadJobTemplate = `apiVersion: batch/v1
+kind: Job
+metadata:
+  name: generate-requests-job
+  namespace: {{.TestNamespace}}
+spec:
+  template:
+    spec:
+      containers:
+      - image: quay.io/zroubalik/hey
+        name: test
+        command: ["/bin/sh"]
+        args: ["-c", "for i in $(seq 1 60);do echo $i;/hey -c 5 -n 80 http://{{.MonitoredAppName}}.{{.TestNamespace}}.svc;sleep 1;done"]
+        securityContext:
+          allowPrivilegeEscalation: false
+          runAsNonRoot: true
+          capabilities:
+            drop:
+              - ALL
+          seccompProfile:
+            type: RuntimeDefault
+      restartPolicy: Never
+  activeDeadlineSeconds: 100
+  backoffLimit: 2
+`
+)
+
+// TestPrometheusScaler creates deployments - there are two deployments - both using the same image but one deployment
+// is directly tied to the KEDA HPA while the other is isolated that can be used for metrics
+// even when the KEDA deployment is at zero - the service points to both deployments
+func TestAzureManagedPrometheusScaler(t *testing.T) {
+	kc := GetKubernetesClient(t)
```

Comment:
Please, add a check to ensure that `TF_AZURE_MANAGED_PROMETHEUS_QUERY_ENDPOINT `has value.
You can check how we do it in other scaler https://github.com/kedacore/keda/blob/main/tests/scalers/azure/azure_application_insights/azure_application_insights_test.go#L150